### PR TITLE
Skipping failing e2e testcases until long term fixed is found.

### PR DESCRIPTION
**Pull request by psbrar99**
Skipping failing e2e testcases until long term fixed is found.

### CODE REVIEW

File: tetrateci/patches/eks/eks-ingress.1.13.patch [hunk 1]
```diff
@@ -0,0 +1,61 @@
+diff --git a/pkg/test/framework/components/echo/kube/deployment.go b/pkg/test/framework/components/echo/kube/deployment.go
```
Comment (incfly) [inline tetrateci/patches/eks/eks-ingress.1.13.patch:1]: How is the patch generated? How will these help the test?

File: tetrateci/test_1.13.sh [hunk 2]
```diff
@@ -31,13 +31,11 @@ if [[ "${CLUSTER}" == "gke" ]]; then
 
   COMMON_TEST_FLAGS+=( "-istio.test.kube.helm.iopFile=${SCRIPTDIR}/iop-gke-integration.yml" )
 
-  echo "Applying GKE specific patches...."
-  git apply "${SCRIPTDIR}/patches/gke/chiron-gke.patch"
```
Comment (incfly) [inline tetrateci/test_1.13.sh:35]: Is this just not needed for 1.13?

**Review by incfly — COMMENTED** [review]
I definitely lack some familiarity of how it works myself, and ask a few questions first.

BTW, could you also reword the PR title to make it clear what this is about and why we do that? "e2e_testcases_wa" is not very informative.

File: tetrateci/patches/eks/eks-ingress.1.13.patch (hunk 1 above)
Comment (incfly) [reply tetrateci/patches/eks/eks-ingress.1.13.patch:1]: I saw a failure in the CI run, https://github.com/tetratelabs/istio/runs/5880304061?check_suite_focus=true in the eks_e2e_test.

```
2022-04-08T10:10:52.0508892Z 2022-04-08T10:10:52.050637Z	warn	tf	failed to get address for port 80: service istio-system/istio-ingressgateway is not available yet: no ingress
```

Is that what this patch trying to solve?

**Comment by psbrar99** [comment]
@incfly : the description was in first comment but let me change the title as you suggested.

File: tetrateci/patches/eks/eks-ingress.1.13.patch (hunk 1 above)
Comment (psbrar99) [reply tetrateci/patches/eks/eks-ingress.1.13.patch:1]: This patch is backported from earlier releases.
For new release we can run `git apply --3way eks_ingress tetrateci/patches/eks/eks-ingress.1.13.patch` > newpatch

Comment (psbrar99) [reply tetrateci/patches/eks/eks-ingress.1.13.patch:1]: @incfly :  regarding failure in CI Run,
`failed to get address for port 80: service istio-system/istio-ingressgateway `
i think this is due to delay in ingress endpoint availabilty, once ip/hostname is assigned by LB it goes away.

File: tetrateci/test_1.13.sh (hunk 2 above)
Comment (psbrar99) [reply tetrateci/test_1.13.sh:35]: i see the test-cases were passing, so did not patched.

2022-04-08T14:16:03.005835Z	info	tf	=== DONE (passed):  Test: 'security_chiron[TestDNSCertificate/rotateDNSCertificatesWhenCertExpired] (5.072950013s)' ===
2022-04-08T14:16:03.005884Z	info	tf	=== DONE (passed):  Test: 'security_chiron[TestDNSCertificate] (15.3082972s)' ===
--- PASS: TestDNSCertificate (15.31s)
    --- PASS: TestDNSCertificate/generateDNSCertificates (0.06s)
    --- PASS: TestDNSCertificate/regenerateDNSCertificates (5.11s)
    --- PASS: TestDNSCertificate/rotateDNSCertificatesWhenCAUpdated (5.07s)
    --- PASS: TestDNSCertificate/rotateDNSCertificatesWhenCertExpired (5.07s)
PASS
2022-04-08T14:16:03.005913Z	info	tf	=== DONE: Test Run: 'security_chiron' ===
2022-04-08T14:16:03.005923Z	info	tf	=== Suite "security_chiron" run time: 45.618731486s ===
2022-04-08T14:16:03.216297Z	info	tf	=== BEGIN: Cleanup Istio [Suite=security_chiron] ===
2022-04-08T14:16:03.216331Z	info	tf	clean up cluster cluster-0
2022-04-08T14:16:34.841214Z	info	tf	=== SUCCEEDED: Cleanup Istio in 31.624826643s [Suite=security_chiron] ===
ok  	istio.io/istio/tests/integration/security/chiron	77.569s

File: tetrateci/1.13/test/skip.d/eks [hunk 3]
```diff
@@ -22,9 +22,9 @@
 #
 # indicates that tests for the package `<pkg>` should not be run at all.
 
-istio.io/istio/tests/integration/helm/upgrade=TestRevisionTagsUpgradeFromPrevious
-MinorRelease
-
+istio.io/istio/tests/integration/helm/upgrade=*
+istio.io/istio/tests/integration/pilot/revisions=*
+istio.io/istio/tests/integration/security/sds_ingress/quic=*
```
Comment (incfly) [inline tetrateci/1.13/test/skip.d/eks:27]: worth commenting why quic* test is skipped. (http3 ingress not avaialble in the testing env)

File: tetrateci/1.13/test/skip.d/gke [hunk 4]
```diff
@@ -20,11 +20,14 @@
 # <pkg>=*
 # ```
 #
-# indicates that tests for the package `<pkg>` should not be run at all.
+# Indicates that tests for the package `<pkg>` should not be run at all.
 
-istio.io/istio/tests/integration/helm/upgrade=TestRevisionTagsUpgradeFromPrevious
-MinorRelease
 
+istio.io/istio/tests/integration/helm=*
```
Comment (incfly) [inline tetrateci/1.13/test/skip.d/gke:26]: trying to speculate some of the failure reasons.

- beta-mtls-off, likely to be a flaky
- remote jwks, might depend on some remote JWK URI endpoint, which causes issue, test not hermetic.
- mtls first party jwt, not quite sure, could be relevant to k8s version.

Could we just quickly check each added skipped test having a legit reason to skip? does not have to be thorough.
thinking as alternative possibility: if something indeed get broken later-on in customer use cases, either caused by some small changes in our own fork, (not much but still some patch, build workflow delta compared with upstream), investigating by then could be difficult.

But feel free to ignore this comment and decide to proceed. I will leave it up to you, to tradeoff. thanks!

File: tetrateci/patches/eks/eks-ingress.1.13.patch (hunk 1 above)
Comment (incfly) [reply tetrateci/patches/eks/eks-ingress.1.13.patch:1]: Thanks. Now I understand that the patch are just copied from previous release. Might worth documenting somewhere in our `docs/*.md`.

**Review by incfly — APPROVED** [review]

**Pull request merged**